### PR TITLE
[ADD] mail_drop_target

### PR DESCRIPTION
**Pull request by hbrunn**
Drag & drop emails to Odoo
==========================

This module was written to allow users to drag&drop emails from their desktop to Odoo.

Usage
=====

To use this module, you need to:

- save your emails on the desktop / somewhere in the file system
- drag them to your browser, and drop them on the chatter of the record you want to attach your email to

Known issues / Roadmap
======================

* most mail clients won't allow you to drag mails directly from the mail client, you'll need some plugin for that
* for corporate environments, it might be feasible to support imap URLs and get the mail in question on the server side

### CODE REVIEW

**Comment by hbrunn** [comment]
depends on https://github.com/OCA/web/pull/907

**Comment by NL66278** [comment]
I was not able to use this in Firefox (60.0.1 64 bit on Linux Mint), Dragging an eml file to either the chatter, or the partner form itself just made the browser tab unusable until CTRL+R to refresh.

**Comment by NL66278** [comment]
Not working in chromium either. One drop actions results in two anonymous entries in the chatter, with no visible content. See screenshot.
2018-06-01-screenprint-chromium-drop-mail-result

**Comment by hbrunn** [comment]
@NL66278 on runbot, this works. I just dragged the notification about your message to runbot, that looks like this: http://3327734-242-27ba7b.runbot2.odoo-community.org/web?#id=1&view_type=form&model=event.event&menu_id=101&action=118 (will die in a while, so please check today)

I guess there's something to do with the integration in the specific project then.

**Review by NL66278 — APPROVED** [review]
After retesting I found out the problem with dragging emails to partners was not due to any problem in the module, but because the file-manager I was using (xfe, otherwise a fantastic tool) does not provide the file in the right way. Dragging from a nemo folder worked as advertised.

File: .travis.yml
Comment (pedrobaeza) [inline .travis.yml:31]: Why not add this to requirements.txt?

File: oca_dependencies.txt
Comment (pedrobaeza) [inline oca_dependencies.txt]: Please remove this and add `web` now that the dependency has been merged

File: .travis.yml
Comment (hbrunn) [reply .travis.yml:31]: because it's optional and you only need it if you joined the dark side: https://github.com/OCA/social/pull/242/files#diff-7030e17fbd71188feaab159c70cc100cR11

Comment (pedrobaeza) [reply .travis.yml:31]: hahaha, OK. I'm from the Jedi side also.

File: oca_dependencies.txt
Comment (hbrunn) [reply oca_dependencies.txt]: done

**Comment by lfreeke** [comment]
:+1: (test)

File: mail_drop_target/models/mail_thread.py
Comment (NL66278) [inline mail_drop_target/models/mail_thread.py]: Unfortunately in newer versions of ExtractMsg, like 0.17, the inner workings of the constructor changed, meaning a StringIO object is no longer valid.:
```
  File "/home/odorboprd11/odoo/parts/runbot/runbot/static/build/05694-10-0-import-development-dc9dd7/parts/odoo/odoo/addons/mail_drop_target/models/mail_thread.py", line 26, in message_process_msg
    message_msg = Message(StringIO(b64decode(message)))
  File "/home/odorboprd11/odoo/parts/runbot/runbot/static/build/05694-10-0-import-development-dc9dd7/eggs/ExtractMsg-0.17-py2.7.egg/ExtractMsg.py", line 657, in __init__
    elif len(path) < 1536:
AttributeError: StringIO instance has no attribute '__len__'
```

Comment (NL66278) [reply mail_drop_target/models/mail_thread.py]: I submitted an issue also on the ExtractMsg project: https://github.com/mattgwwalker/msg-extractor/issues/44

Comment (hbrunn) [reply mail_drop_target/models/mail_thread.py]: thanks, I removed the StringIO call, if not much more changes, we'll be fine.

There wasn't any specific reason to choose StringIO, that simply was what the documentation said so I went with it.

Comment (NL66278) [reply mail_drop_target/models/mail_thread.py]: Retested and works fine now!

**Review by etobella — APPROVED** [review]
I have tested it and it works like a charm!!!
:+1:

**Comment by etobella** [comment]
@hbrunn Can you rebase your commits? I would like to migrate it to 11.0 as I think it is a great idea :smile:

**Comment by hbrunn** [comment]
@etobella just squash them, the commit hash will change after rebase+merge here anyways. Please write a test for the python part in your port, then I can grab it for this PR to get merged

**Comment by etobella** [comment]
@hbrunn I added some tests in order to check that all worked fine. Can you cherry-pick them in order to have this PR ready to merge?

**Comment by hbrunn** [comment]
sure, thanks, will do next time I work on this. If you're in a hurry, just PR them to my branch and I'll merge today

**Comment by hbrunn** [comment]
but there's nothing wrong with merging your 11.0 version first without this PR being merged at all for the time being

**Comment by etobella** [comment]
@hbrunn I know it is OK, I was trying to remind you about it :wink:

**Review by rafaelbn — APPROVED** [review]
:+1:

**Comment by hbrunn** [comment]
may I now instruct mergebot myself to merge this? Or if not, would someone else do the honors?

**Comment by pedrobaeza** [comment]
Maybe some squashing?

**Comment by hbrunn** [comment]
I think it would be rude to squash away @etobella's commit. Then there was one he cherry picked to 11, so I better keep this intact too. Leaves the ones between those, and they can't be squashed...

**Pull request merged**
